### ISSUE TITLE
Migrate S3 bucket to hold Bazel cache to a MCF AWS account

### ISSUE THREAD

**Issue by themarwhal** [magma/magma#12058]
The existing Magma VM/Decontainer cache used for developement is current hosted by FB-magma account owned by Meta. This issue tracks the work of migrating this bucket to live in the LF / MCF AWS account.

**Comment by jheidbrink** [comment]
From https://docs.aws.amazon.com/AmazonS3/latest/userguide/BucketRestrictions.html

> You cannot transfer bucket ownership to another AWS account

If we want to keep the bucket name (`magma-cache`), we could delete the bucket and create a new one with that name in a new AWS account. The documentation says “some time might pass before you can reuse the name of a deleted bucket”. For a seamless migration, we probably don't want to wait "some time" before we can create the new bucket, so it's probably easiest to create a new bucket with a different name.

I think the migration could then be done with the following steps (I annotated those that need elevated privileges in italics):

* Create a new bucket in the LF / MCF AWS account and a corresponding IAM role (name suggestion: "mcf-magma-cache").
  * requires _access to the LF / MCF AWS account_
* Store the access key for the new IAM role in Github Secrets
  * requires _the access key for the new IAM role_, and _admin access to the Magma Github repository_
* optional: Copy Bazel cache files from `s3://magma-cache` to the new bucket (we could also rebuild the cache, but that takes > 2 hours as opposed to some minutes for copying)
  * requires _the access key for the new IAM role in the LF/MCF AWS account_
* confirm that files in the new bucket are publicly readable
* Adapt `github/workflows/bazel-cache-push.yml` to upload to the new bucket using the new access key
* remove the access key for the old bucket from the secrets of the Magma Github repository
  * requires _admin access to the Magma Github repository_
* Adapt `devcontainer/post-create-commands.sh` to use the new bucket.
* Wait until the changes have propagated to all relevant branches (maybe actively backport?)
* Delete the old magma-cache bucket
  * requires _access to the FB-magma AWS account_

**Comment by themarwhal** [comment]
> From https://docs.aws.amazon.com/AmazonS3/latest/userguide/BucketRestrictions.html
> 
> 
> 
> > You cannot transfer bucket ownership to another AWS account
> 
> 
> 
> If we want to keep the bucket name (`magma-cache`), we could delete the bucket and create a new one with that name in a new AWS account. The documentation says “some time might pass before you can reuse the name of a deleted bucket”. For a seamless migration, we probably don't want to wait "some time" before we can create the new bucket, so it's probably easiest to create a new bucket with a different name.
> 
> 
> 
> I think the migration could then be done with the following steps (I annotated those that need elevated privileges in italics):
> 
> 
> 
> * Create a new bucket in the LF / MCF AWS account and a corresponding IAM role (name suggestion: "mcf-magma-cache").
> 
>   * requires _access to the LF / MCF AWS account_
> 
> * Store the access key for the new IAM role in Github Secrets
> 
>   * requires _the access key for the new IAM role_, and _admin access to the Magma Github repository_
> 
> * optional: Copy Bazel cache files from `s3://magma-cache` to the new bucket (we could also rebuild the cache, but that takes > 2 hours as opposed to some minutes for copying)
> 
>   * requires _the access key for the new IAM role in the LF/MCF AWS account_
> 
> * confirm that files in the new bucket are publicly readable
> 
> * Adapt `github/workflows/bazel-cache-push.yml` to upload to the new bucket using the new access key
> 
> * remove the access key for the old bucket from the secrets of the Magma Github repository
> 
>   * requires _admin access to the Magma Github repository_
> 
> * Adapt `devcontainer/post-create-commands.sh` to use the new bucket.
> 
> * Wait until the changes have propagated to all relevant branches (maybe actively backport?)
> 
> * Delete the old magma-cache bucket
> 
>   * requires _access to the FB-magma AWS account_

This looks great! Thank you for writing up the steps. The new name looks good to me. We only care about the master branch for this, so probably no need for back ports. As far as I know, the bucket is only being used actively by the DevContainer startup script.

**Comment by jheidbrink** [comment]
i pinged Kendall on Slack, and she filed a ticket with LF IT for the AWS account. I'll proceed with these steps once we are good to go! 

> As far as I know, the bucket is only being used actively by the DevContainer startup script.

Yes, that was also the only place that I found.

**Comment by jheidbrink** [comment]
Superseded by #12210
